Fixes error where app module fails to load in testing

kanhirun opened the pull request:
# Summary
- Fixes a known compatibility issue where the app module fails to load in the testing environment (5fa13245cc536af723a464cb02bc9b2d54ea5b80, https://github.com/angular/angular.js/issues/13794)
- Fixes an existing test to be more robust: previously, the assertion would pass on _every_ error which is undesirable. [[LOC](https://github.com/CrossLead/angular-zendesk-widget/pull/4/commits/5fa13245cc536af723a464cb02bc9b2d54ea5b80#diff-1d1c767b303c6751bae1b1c77086c713R11)]
- Because Karma doesn't officially support PhantomJS, this PR removes its support in favor of Chrome (5ad766407e53ac6ae64f6b8e65661f1727af45cf) If you strongly prefer PhantomJS, we can alternatively fix the issue by upgrading to PhantomJS 2.x. [[Review](https://github.com/CrossLead/angular-zendesk-widget/commit/1162b724ddaccbb181a25298d40dc91f8990fbba)]. I'll leave it to you to decide.
## Tasks
- [x] Comment below if you prefer upgrading PhantomJS
- [x] If this PR is merged, delete the alternative branch: `git push origin --delete  fixes_ng_module_error_phantomjs_upgrade`
## Testing
- [x] Run `./node_modules/.bin/gulp test-src`. Is the test passing?
- [x] Change the [assertion message](https://github.com/CrossLead/angular-zendesk-widget/pull/4/files#diff-1d1c767b303c6751bae1b1c77086c713R11) to purposely fail the test , run the test, and observe that it fails with the correct feedback

kanhirun committed:
Fixes error where app module cannot be imported

When calling `module()` with `inject()`, an error is thrown:
```
Error: [$injector:modulerr] Failed to instantiate module ng due to:
TypeError: 'undefined' is not an object (evaluating
'Function.prototype.bind.apply')
```

PhantomJS is not amongst Karma's officially supported browsers.
In 1.x, `Function.prototype.bind` isn't defined. Instead of polyfilling,
or using Phantom 2.x, it is better to stick with supported browsers.

(In addition, TravisCI has been reconfigured to run Chrome.)

READ MORE:
http://blog.500tech.com/setting-up-travis-ci-to-run-tests-on-latest-google-chrome-version/

kanhirun commented:
@bsouthga Ready for review. I've included checkboxes on the PR to test on your machine.

kanhirun committed:
Chore: Removes PhantomJS support on Karma

READ MORE: 5fa13245cc536af723a464cb02bc9b2d54ea5b80

bsouthga commented:
Hey kel, thanks for the PR! really appreciate your detailed description of the issue + your fix. I merged in a travis hook to test this with our CI setup and it seems to fail (as there is no headless chrome available, the one advantage of Phantom) -- any thoughts on a fix for this? We would love to keep the ability to run the automated tests if possible.

kanhirun committed:
Fixes existing tests to successfully run as minified

The matcher, `toThrowError()` is valid only in Jasmine, and not Mocha.
Managing multiple competing test frameworks is not ideal. Therefore, I
have configured all karma config files to use Jasmine only.

kanhirun committed:
Chore: Replaces support for Mocha for Jasmine

See: 8e89dfa4af42b4a3dbae7583dd15537ce621e753

kanhirun commented:
### New Changes

@bsouthga I've added two commits recently:
- Fixes existing tests to run as both concatenated and minified (a8138597988777905b9f5fbb5e121dee3e1b91ee)
- Because managing competing test frameworks is complex, I've removed support for Mocha in favor of Jasmine. This is preferable because—although the two frameworks have very similar DSL—there are still many possible leaks. For example, `toThrowError()` is a matcher only valid in Jasmine. (a9da0fcece8637b65db36a4614348e52112fe615)

### Additional Testing
- [ ] `./node_modules/bin/gulp test-dist-concatenated` is now passing when previously it had not (see: dfee8b2880876cd1edb5c8936698da0875234c2e)
- [ ] `./node_modules/bin/gulp test-dist-minified` is now passing as well

I'll look into configuring TravisCI for running Chrome. Thanks for bringing this up. 👍

kanhirun committed:
Adds missing build artifacts

See: 2a2a03173c6e240111c5ea84c50323ce34e1e4aa

kanhirun commented:
@bsouthga 🎱  says that it's now passing. I squashed those changes on `.travis.yml` into 470a8e8093fba9a3622ec6f833a438ba075ad384 (the first commit) so that [git-bisect](https://git-scm.com/docs/git-bisect) stays effective.

bsouthga commented:
Looks good to me!
